src/test/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepositoryIT.java

```diff
@@ -58,140 +75,419 @@
+            assertNotNull(response);
+            // verify response can deserialize at least the items supplied (and a select few others)
+            assertNotNull(response.getRoleAssignmentRequest());
+            // :: verify header
+            assertNotNull(response.getRoleAssignmentRequest().getRequest());
+            RoleRequestResource roleRequestHeader = response.getRoleAssignmentRequest().getRequest();
+            RoleRequestResource suppliedRoleRequestHeader = assignmentRequest.getRequest();
+            assertAll(
+                () -> assertThat(roleRequestHeader.getAssignerId(), is(suppliedRoleRequestHeader.getAssignerId())),
+                () -> assertThat(roleRequestHeader.getProcess(), is(suppliedRoleRequestHeader.getProcess())),
+                () -> assertThat(roleRequestHeader.getReference(), is(suppliedRoleRequestHeader.getReference())),
+                // additional fields populated from RAS
+                () -> assertNotNull(roleRequestHeader.getRequestType()),
+                () -> assertNotNull(roleRequestHeader.getStatus()),
+                () -> assertNotNull(roleRequestHeader.getCreated())
+            );
+            // :: verify roles
+            assertThat(
+                response.getRoleAssignmentRequest().getRequestedRoles().size(),
+                is(assignmentRequest.getRequestedRoles().size())
+            );
+
+            Map<String, RoleAssignmentResource> roleMap = response.getRoleAssignmentRequest().getRequestedRoles()
+                .stream().collect(Collectors.toMap(RoleAssignmentResource::getRoleName, role -> role));
+
+            assignmentRequest.getRequestedRoles().forEach(suppliedRequestedRole -> {
+                String roleName = suppliedRequestedRole.getRoleName();
+                assertAll(
+                    () -> assertTrue(roleMap.containsKey(roleName)),
+                    () -> validateRoleAssignment(roleMap.get(roleName), suppliedRequestedRole)
+                );
+            });
+        }
+
+        @DisplayName("should error on 400 when POST roleAssignments")
+        @Test
+        void shouldErrorOn400WhenCreateRoleAssignment() {
+
+            // GIVEN
+            RoleAssignmentRequestResource assignmentRequest = createAssignmentRequest(Set.of("[ROLE1]"));
+            badRasStub = WireMock.stubFor(WireMock.post(urlMatching("/am/role-assignments")).willReturn(badRequest()));
+
+            // WHEN / THEN
+            final BadRequestException exception = assertThrows(BadRequestException.class,
+                () -> roleAssignmentRepository.createRoleAssignment(assignmentRequest));
+
+            assertThat(exception.getMessage(), startsWith(
+                    String.format(DefaultRoleAssignmentRepository.ROLE_ASSIGNMENTS_CLIENT_ERROR, "creating", "")
+                )
+            );
+        }
+
+        @DisplayName("should error on 500 when POST roleAssignments")
+        @Test
+        void shouldErrorOn500WhenCreateRoleAssignment() {
```

Comment:
rename test: `shouldErrorServiceExceptionWhenCreateRoleAssignmentReturnsServerError`

Reply:
Resolved in commit cd1aaacf05d0591bb6b7338046d4c07073b49139.